starfish/codebook.py

```diff
@@ -442,40 +456,45 @@ def metric_decode(
         Returns
         -------
         IntensityTable :
-            intensity table containing additional data variables for target assignments and metric
-            outputs
+            Intensity table containing normalized intensities, target assignments, distances to
+            the nearest code, and the filtering status of each feature.
 
         """
+
+        self._validate_decode_intensity_input_matches_codebook_shape(intensities)
+
         # normalize both the intensities and the codebook
         norm_intensities, norms = self._normalize_features(intensities, norm_order=norm_order)
         norm_codes, _ = self._normalize_features(self, norm_order=norm_order)
 
-        # mask low intensity features
-        intensity_mask = np.where(norms < min_intensity)[0]
-
         metric_outputs, targets = self._approximate_nearest_code(
             norm_codes, norm_intensities, metric=metric)
 
-        exceeds_distance = np.where(metric_outputs > max_distance)[0]
-        target_index = pd.Index(targets)
+        passes_filters = np.ones_like(targets, dtype=np.bool)
 
-        # remove targets associated with distant codes
-        target_index.values[exceeds_distance] = 'None'
-        target_index.values[intensity_mask] = 'None'
+        # indicate that targets with high hamming distances or low intensities should be filtered
+        intensity_mask = norms < min_intensity
+        distance_mask = metric_outputs > max_distance
+        passes_filters[intensity_mask | distance_mask] = False
 
-        # set new values on the intensity table in-place
-        intensities[Features.TARGET] = (Features.AXIS, target_index)
-        intensities[Features.DISTANCE] = (Features.AXIS, metric_outputs)
+        # set targets, distances, and filtering results
+        norm_intensities[Features.TARGET] = (Features.AXIS, targets)
+        norm_intensities[Features.DISTANCE] = (Features.AXIS, metric_outputs)
+        norm_intensities[Features.PASSES_FILTERS] = (Features.AXIS, passes_filters)
 
-        return intensities
+        # norm_intensities is a DataArray, make it back into an IntensityTable
+        return IntensityTable(norm_intensities)
```

Comment:
any perf implications for this?

Reply:
test if mutating one data set changes the other.

Reply:
![image](https://user-images.githubusercontent.com/5239152/45222513-cc6d2e00-b282-11e8-8c98-84961b496730.png)

View, not copy. I think we're OK. 

I didn't add a test to the suite, since we would have to return a non-`IntensityTable` to do so.